system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -62,10 +63,13 @@ export class _TopSites extends React.PureComponent {
   }
 
   /**
-   * Return the TopSites to display based on prefs.
+   * Return the TopSites that are visible based on prefs and window width.
    */
-  _getTopSites() {
-    return this.props.TopSites.rows.slice(0, this.props.TopSitesRows * TOP_SITES_MAX_SITES_PER_ROW);
+  _getVisibleTopSites() {
+    // We hide 2 sites per row when not in the wide layout.
+    const WIDE_LAYOUT_BREAKPOINT = 1056;
+    const sitesPerRow = this.windowObj.innerWidth < WIDE_LAYOUT_BREAKPOINT ? TOP_SITES_MAX_SITES_PER_ROW - 2 : TOP_SITES_MAX_SITES_PER_ROW;
```

Comment:
Given that we should have a live document, there should be a dynamic way to query what's shown… I'll think about it a bit more..